lib/caching/inMemoryNonceCacheProvider.ts

```diff
@@ -47,7 +47,7 @@ export class InMemoryNonceCacheProvider implements INonceCacheProvider {
         return { lockAquired: true, data: nonceData };
       }
 
-      return { lockAquired: false, data: undefined };
+      return { lockAquired: false, data: nonceData };
```

Comment:
@alatras @adriancodrutd We need to update the redis cache implementation to return the nonceData even if a lock cannot be aquired.